src/components/AppBar/Drawer.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import Drawer from '@mui/material/Drawer';
+import List from '@mui/material/List';
+import Divider from '@mui/material/Divider';
+import ListItem from '@mui/material/ListItem';
+import ListItemButton from '@mui/material/ListItemButton';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import ListItemText from '@mui/material/ListItemText';
+import IconButton from '@mui/material/IconButton';
+import SailingIcon from '@mui/icons-material/Sailing';
+import Avatar from '@mui/material/Avatar';
+import LogoutIcon from '@mui/icons-material/Logout';
+import SportsEsportsIcon from '@mui/icons-material/SportsEsports';
+import ForumIcon from '@mui/icons-material/Forum';
+
+export default () => {
+  const [isOpened, setIsOpened] = React.useState(false)
+
+  const toggleDrawer = (open: boolean) => (event: React.KeyboardEvent | React.MouseEvent) => {
+    if (
+      event.type === 'keydown'
+      && ((event as React.KeyboardEvent).key === 'Tab'
+      || (event as React.KeyboardEvent).key === 'Shift')
+    ) { return; }
+
+    setIsOpened(open)
+  }
+
+  const goToProfile = () => {
```

Comment:
❗️
`useCallback`, здесь и везде

Reply:
А зачем? Ориентируюсь на этот пример https://www.w3schools.com/react/react_usecallback.asp :

> Every time a component re-renders, its functions get recreated

> we can use the useCallback hook to prevent the function from being recreated unless necessary

Т.е смысл в такой оптимизации был бы при наличии в `Drawer.tsx` хотя бы еще одного `state variable` не связанного с закрыванием/открыванием меню. В нашей же ситуации имеется единственная `state variable`, а значит мы не можем избежать пересоздания функций.

Reply:
Они прокидываются пропсами. Если они прокидываются в чистые компоненты, то referential equality нужна для оптимизации